### PR TITLE
Add "Size" column to entry view

### LINKED ISSUE #4101: Add column "Size" so a Search by Size is possible
## Summary
Show my password entries sorted by date or by size

## Details
I suggested these two views to KeePass and Dominik implemented them:

The following screen shot is the "Find" menu in KeePass:
image

This shows all entries in my password database in a new window sorted by size or sorted by date.

**Searching by size** is very useful when you want to find large entries.
I had one entry with 4 MB in size, it contained a PDF (1 MB) and the same PDF 3 times in the history, So I delete the history and replaced the PDF with a JPG, now the entry is 60 KB.

**Searching by size** is very useful when you want to see what entries you have last modified (I use this very often).

### PR DESCRIPTION
**Pull request by sigmaupsilon**
## Type of change
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
Adds a size column to the entry view that displays the size of each entry in KiB

Closes #4101

## Screenshots
sizecolumn

## Testing strategy
Unit and Manual Testing

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ I have modified tests to cover my changes.

### CODE REVIEW

File: src/core/Entry.cpp [hunk 1]
```diff
@@ -1202,6 +1219,7 @@ Entry::PlaceholderType Entry::placeholderType(const QString& placeholder) const
         {QStringLiteral("{PASSWORD}"), PlaceholderType::Password},
         {QStringLiteral("{NOTES}"), PlaceholderType::Notes},
         {QStringLiteral("{TOTP}"), PlaceholderType::Totp},
+        {QStringLiteral("{SIZE}"), PlaceholderType::Size},
```
Comment (droidmonkey) [inline src/core/Entry.cpp:1222]: There is no need for a reference placeholder. Plus size is not an attribute.

File: src/core/EntrySearcher.cpp [hunk 2]
```diff
@@ -181,12 +183,16 @@ bool EntrySearcher::searchEntryImpl(Entry* entry)
             found = entry->attributes()->contains(term.word)
                     && term.regex.match(entry->attributes()->value(term.word)).hasMatch();
             break;
+        case Field::Size: {
+            found = term.regex.match(entrySizeKiB).hasMatch();
```
Comment (droidmonkey) [inline src/core/EntrySearcher.cpp:187]: Searching by size is only useful if we have < and > defined. Please remove the searcher entries. The more useful approach for a user is to search by `*` and sort the size column largest to smallest.

File: src/core/EntrySearcher.cpp [hunk 3]
```diff
@@ -149,6 +149,8 @@ bool EntrySearcher::searchEntryImpl(Entry* entry)
     auto attributes = QStringList(attributes_keys + entry->attributes()->values(attributes_keys));
     auto attachments = QStringList(entry->attachments()->keys());
 
+    QString entrySizeKiB = QString::number(entry->size() / 1024.0, 'f', 2);
```
Comment (droidmonkey) [inline src/core/EntrySearcher.cpp:152]: KiB is kind of obscure for most people, recommend just KB. Additionally, the size value should scale with the size, similar to a file system. If the size is over 1000 KB then it's time to show 1 MB. Same with B, prefer to not have 0.08 KB, should be 80 B.

Comment (phoerious) [reply src/core/EntrySearcher.cpp]: Actually, I would prefer KiB simply for the reason that it tends to be the default unit on all operating systems. Only hard disk manufacturers use base 1000.

File: src/core/Entry.cpp (hunk 1 above)
Comment (sigmaupsilon) [reply src/core/Entry.cpp]: Removed placeholder code

File: src/core/EntrySearcher.cpp (hunk 2 above)
Comment (sigmaupsilon) [reply src/core/EntrySearcher.cpp]: Removed search by size

File: src/core/EntrySearcher.cpp (hunk 3 above)
Comment (sigmaupsilon) [reply src/core/EntrySearcher.cpp]: Sizes have been reverted to B, KiB, etc.

**Comment by phoerious** [comment]
Please rebase to develop.

**Comment by sigmaupsilon** [comment]
Thank you, I have updated the my branch to develop

**Comment by phoerious** [comment]
Something is wrong with the CLI test on Linux. I though it was just a random CI failure, but it is very consistent and I could also produce it locally. I would like to ask you to look into that.

**Comment by phoerious** [comment]
Actually, never mind. The same happens on develop. I am not sure what went wrong there.

**Comment by lshman** [comment]
I also had a problem with the cli test. It turned out I didn't have the xclip program installed. After installing xclip the cli test worked fine

File: src/gui/entry/EntryModel.cpp [hunk 4]
```diff
@@ -222,6 +222,22 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
...
         }
+        case Size: {
+            const int unitsSize = 4;
+            QString units[unitsSize] = {"B", "KiB", "MiB", "GiB"};
+            float resultInt = entry->size();
+
+            for (int i = 0; i < unitsSize; i++) {
+                if (resultInt < 1024 || i == unitsSize - 1) {
+                    resultInt = qRound(resultInt * 100) / 100.0;
+                    result = QString::number(resultInt) + " " + units[i];
```
Comment (droidmonkey) [inline src/gui/entry/EntryModel.cpp:233]: Use: `QStringLiteral("%1 %2").arg(QString::number(resultInt), units[i])`

**Pull request merged**